### ISSUE TITLE
JSON display shows u"..."

### ISSUE THREAD

**Issue by wbolster** [amjith/pgcli#163]
When `pgcli` is run in Python 2, JSON fields (`json` and `jsonb` types in PostgreSQL), are shown with `u"..."` prefixes for strings. I guess this is a side-effect of using `repr()`. Maybe the `pprint` module can be leveraged instead?

**Comment by amjith** [comment]
The code doesn't use `repr()` at least not explicitly, so I'll have to find out where the unicode prefix is coming from. 

I don't have any table with json or jsonb, but I do have a table with `hstore` and it doesn't seem to have the 'u' prefix on `hstore`. I'll create a reproducible case and give it as shot tonight. 

Thanks for reporting.

**Comment by drocco007** [comment]
Here is a simple case that demonstrates the issue:

    _test_db> CREATE TABLE json_test (id serial, data JSON);           
    CREATE TABLE
    Command Time: 5.00679016113e-06
    Format Time: 2.31266021729e-05
    _test_db> INSERT INTO json_test (data) VALUES ('{"name": "Éowyn"}')
    INSERT 0 1
    Command Time: 3.2901763916e-05
    Format Time: 3.40938568115e-05
    _test_db> SELECT data FROM json_test                               
    +------------------------+
    | data                   |
    |------------------------|
    | {u'name': u'\xc9owyn'} |
    +------------------------+
    SELECT 1
    Command Time: 2.69412994385e-05
    Format Time: 0.000692844390869
    _test_db>

It appears that psycopg2 is causing this: recent versions by default run json data through `json.loads` on the way out of the database: http://initd.org/psycopg/docs/extras.html#json-adaptation

Adapting psycopg2's recipe and placing the following top of `pgexecute.py` fixes this case:

    psycopg2.extras.register_default_json(loads=lambda x: x.decode('utf-8'))

which renders the output correctly:

    _test_db> SELECT data FROM json_test                               
    +-------------------+
    | data              |
    |-------------------|
    | {"name": "Éowyn"} |
    +-------------------+
    SELECT 1
    Command Time: 4.05311584473e-06
    Format Time: 0.000294923782349
    _test_db> 

However, this probably only works if the database encoding is in fact UTF-8; it may throw decode errors for other encodings. If you'd like, I can submit a PR that uses the database connection's encoding (which seems to be what psql does).

**Comment by amjith** [comment]
Interesting. I thought psycopg2 treated everything that came out of the database as UTF-8. 

That's why I had to explicitly tell it to not convert unicode to UTF-8 using these register_type() calls. https://github.com/amjith/pgcli/blob/master/pgcli/pgexecute.py#L13..L16

I'll gladly accept a PR that uses the database's encoding. :)

**Comment by amjith** [comment]
Fix is now available in master. 

You can verify the fix by installing directly from the git repo as follows: 

```
pip install -U -e git+https://github.com/amjith/pgcli.git#egg=pgcli
```

**Comment by wbolster** [comment]
Thanks!  Fwiw, the problem I reported basically boiled down to this (Python 2 only):

```python
>>> d = {u'foo': u'bar'}
>>> str(d)
"{u'foo': u'bar'}"
```

Before this issue was addressed `pgcli` showed a `str()` representation of a Python dict, which happens to look like JSON (but isn't JSON). After the change, the JSON string as it is sent by the DB is shown. This will work as long as `pgcli` doesn't to anything with the JSON result. I can imagine pretty-printing it   (especially in combination with #162) would make a lot of sense... ;)
